src/server/scripts/EasternKingdoms/ShadowfangKeep/boss_apothecary_hummel.cpp

```diff
@@ -0,0 +1,587 @@
+            {
+                GetHitUnit()->CastSpell(GetHitUnit(), SPELL_COLOGNE_SPILL, true);
+            }
+
+            void Register() override
+            {
+                OnEffectHitTarget += SpellEffectFn(spell_apothecary_throw_cologne_SpellScript::HandleScript, EFFECT_0, SPELL_EFFECT_DUMMY);
+            }
+        };
+
+        SpellScript* GetSpellScript() const override
+        {
+            return new spell_apothecary_throw_cologne_SpellScript();
+        }
+};
+
+// 68966 - Throw Perfume
+class spell_apothecary_throw_perfume : public SpellScriptLoader
+{
+    public:
+        spell_apothecary_throw_perfume() : SpellScriptLoader("spell_apothecary_throw_perfume") { }
+
+        class spell_apothecary_throw_perfume_SpellScript : public SpellScript
+        {
+            PrepareSpellScript(spell_apothecary_throw_perfume_SpellScript);
+
+            void HandleScript(SpellEffIndex /*effindex*/)
+            {
+                GetHitUnit()->CastSpell(GetHitUnit(), SPELL_ALLURING_PERFUME_SPILL, true);
+            }
+
+            void Register() override
+            {
+                OnEffectHitTarget += SpellEffectFn(spell_apothecary_throw_perfume_SpellScript::HandleScript, EFFECT_0, SPELL_EFFECT_DUMMY);
+            }
+        };
+
+        SpellScript* GetSpellScript() const override
+        {
+            return new spell_apothecary_throw_perfume_SpellScript();
+        }
+};
+
+// 68798 - Concentrated Alluring Perfume Spill
+class spell_apothecary_perfume_spill : public SpellScriptLoader
+{
+    public:
+        spell_apothecary_perfume_spill() : SpellScriptLoader("spell_apothecary_perfume_spill") { }
+
+        class spell_apothecary_perfume_spill_AuraScript : public AuraScript
+        {
+            PrepareAuraScript(spell_apothecary_perfume_spill_AuraScript);
+
+            void OnPeriodic(AuraEffect const* /*aurEff*/)
+            {
+                GetTarget()->CastSpell((Unit*)nullptr, SPELL_PERFUME_SPILL_DAMAGE, true);
+            }
+
+            void Register() override
+            {
+                OnEffectPeriodic += AuraEffectPeriodicFn(spell_apothecary_perfume_spill_AuraScript::OnPeriodic, EFFECT_0, SPELL_AURA_PERIODIC_DUMMY);
+            }
+        };
+
+        AuraScript* GetAuraScript() const override
+        {
+            return new spell_apothecary_perfume_spill_AuraScript();
+        }
+};
+
+// 68614 - Concentrated Irresistible Cologne Spill
+class spell_apothecary_cologne_spill : public SpellScriptLoader
+{
+    public:
+        spell_apothecary_cologne_spill() : SpellScriptLoader("spell_apothecary_cologne_spill") { }
+
+        class spell_apothecary_cologne_spill_AuraScript : public AuraScript
+        {
+            PrepareAuraScript(spell_apothecary_cologne_spill_AuraScript);
+
+            void OnPeriodic(AuraEffect const* /*aurEff*/)
+            {
+                GetTarget()->CastSpell((Unit*)nullptr, SPELL_PERFUME_SPILL_DAMAGE, true);
```

Comment:
The damage from the spill is working? can you post a Screenshot please? Even changing targets, i can't see any damage here :/